### PR TITLE
Reorganise & Update Blocks Documentation for v1.20.20 (Skipped)

### PR DESCRIPTION
**Pull request by QuazChick**
Merge along with https://github.com/Bedrock-OSS/wiki-addon/pull/20 when 1.20.20 releases.

## Block Features

Over time, the block features page has become very long, being home to components, states, permutations and now traits documentation. To ease accessibility, the following changes have been made.

image

- Block features page removed and split into 5 new pages (sorted into learning order):
  - ### Block Components
    - Added application example.
    - Now includes `minecraft:unit_cube` - marked as experimental.
    - Render method information from `block-materials.md` moved to `Material Instances` section.
      - Removed Block Materials page.
    - Headings are now more readable to better support the table of contents e.g.,
      `minecraft:crafting_table` -> `Crafting Table`.
  - ### Block States
    - Now contains more information about getting and setting states.
  - ### Block Traits
  - ### Block Permutations
    - Added section explaining condition Molang limitations.
  - ### Block Events & Triggers
    - Expanded documentation of the `play_effect` and `play_sound` event responses.

- Each feature page has a description defined.
- Added dead link redirects to 404 page:
  - `blocks/blocks-16` -> `blocks/blocks-intro`
  - `blocks/blocks-stable` -> `blocks/blocks-intro`
  - `blocks/block-materials` -> `blocks/block-components#render-methods`

## Experimental Blocks
- Experimental feature documentation has been moved to its relevant pages.
- This page now acts as an index of experimental requirements:
  image

## Intro to Blocks
- Basic block example no longer requires `minecraft:unit_cube` which was previously unmentioned.
- Removed flipbook textures as they have their own dedicated page.
- Added new example block textures.
- Added "What's Next?" card section:
  image

## Block Tutorials
- Updated to format & min engine versions `1.20.20`.
  - Converted block `properties` to `states`.
- Renamed `custom-trapdoor.md` to `custom-trapdoors.md` for consistency.
- Code snippets are no longer incorrectly wrapped in `{}`, making the "Copy" function more practical.

## Home Page
- Update blocks card to link to block components.
- Correct grammatical errors.

### CODE REVIEW

File: docs/blocks/applying-effects.md [hunk 1]
```diff
@@ -24,91 +24,79 @@ This tutorial aims to show how to apply status effects to entities as long as th
 
 ## Setup
 
-We will need to add a couple things to our code, first let's start with property that will hold `true` and `false` values:
+We will need to add a couple things to our code, first let's start with a state that will hold `true` when stood on, and `false` otherwise:
 
-<CodeHeader>BP/blocks/wither_block.json</CodeHeader>
+<CodeHeader>minecraft:block > description</CodeHeader>
```
Comment (MedicalJewel105) [inline docs/blocks/applying-effects.md:29]: Before we always did BP/blocks/wither_block.json#description. Probably your method is better.

File: docs/blocks/block-sounds.md
Comment (MedicalJewel105) [inline docs/blocks/block-sounds.md:1]: Bedrock wiki content generator will wipe all what you've typed. I'll update it so it generates a table, not a list.

File: docs/blocks/block-states.md
Comment (MedicalJewel105) [inline docs/blocks/block-states.md:1]: Usually we don't use tabulation (spaces) for ##, ### and ####.

File: docs/blocks/blocks-stable.md
Comment (MedicalJewel105) [inline docs/blocks/blocks-stable.md:1]: I thought there somewhere should be a page with non experimental block components. Some marketplace creators use our wiki too and they can not use experiments as IK.

Comment (QuazChick) [reply docs/blocks/blocks-stable.md:1]: I've made sure to clearly mark if one section of a page is experimental (currently only `minecraft:unit_cube` is under experimental):
image

The entirety of `block-events` - which contains `minecraft:on_*` and ticking trigger components - is marked as experimental:
image

There are also simplified experimental tables in `blocks-experimental`:
image

File: docs/blocks/block-states.md
Comment (QuazChick) [reply docs/blocks/block-states.md:1]: It was for list formatting, however it looks almost identical without so I'm not sure why I made it a list. The list formatting has now been removed!

File: docs/blocks/troubleshooting-blocks.md
Comment (MedicalJewel105) [inline docs/blocks/troubleshooting-blocks.md:1]: Why you removed navigation order?

**Review by MedicalJewel105 — COMMENTED** [review]
Looks good!

File: docs/blocks/troubleshooting-blocks.md
Comment (QuazChick) [reply docs/blocks/troubleshooting-blocks.md:1]: It didn't have any effect, it's automatically placed into the right position - after all other pages in the `General` category.

**Pull request closed without merging**